config/gulp/translation.js

```diff
@@ -0,0 +1,71 @@
+var gulp = require('gulp');
+var gutil = require('gulp-util');
+var del = require('del');
+var pkg = require('../../package.json');
+var runSequence = require('run-sequence');
+var spawn = require('cross-spawn');
+var rename = require('gulp-rename');
+var replace = require('gulp-replace');
+var stateNames = require('./state-names.json');
+
+gulp.task('clean-translation', function () {
+  gutil.log(gutil.colors.cyan('clean-translation'), 'Removing generated register/ files');
+  return del([
+    './layouts/registrar',
+    './content/registrar',
+  ]);
+});
+
+gulp.task('copy-content-spanish', function (done) {
+
+  return gulp.src('./content/register/*.md')
+    .pipe(rename(function (path) {
+      var file = stateNames[path.basename].file;
+      path.basename = file;
+    }))
+    .pipe(replace(/title = "(.+)"/, function (match, p1) {
+      var name = p1.replace(/\s/g, '-').replace(/\./g, '').toLowerCase();
+      var title = stateNames[name].title;
+      var file = stateNames[name].file;
+      return (
+        'title = "' + title + '"' +
+        "\n" +
+        'slug = "' + file + '"'
+      );
+    }))
+    .pipe(gulp.dest('./content/registrar'));
+
+});
+
+gulp.task('copy-layouts-spanish', function (done) {
+
+  var copyLayout = spawn('cp', [
+    '-rvf',
+    './layouts/register',
+    './layouts/registrar',
+  ]);
+
+  copyLayout.stdout.on('data', function (data) {
+    gutil.log(gutil.colors.blue('copy-layouts-spanish'), '\n' + data);
+  });
+
+  copyLayout.on('error', done);
+  copyLayout.on('close', done);
+
+
+});
+
+gulp.task('copy-translation', [ 'clean-translation' ], function (done) {
+
+  gutil.log(
+    gutil.colors.cyan('copy-translation'),
+    'Copying files from content/ & layouts/ for translated URLs'
+  );
+
+  runSequence(
+    'copy-content-spanish',
+    'copy-layouts-spanish',
+    done
+  );
+
+});
```

Comment:
This copy of files is necessary _just to build the translated URLs_ because Hugo uses the file-system to determine what URLs to use. There isn't a good way to combine these into a single `content/**/*.md` file for each state and territory because the `slug` property doesn't allow for an array of URLs. The `slug` property only allows a single string. We don't want to redirect on the spanish / english URLs since we want actual translated URLs for each state / territory.

### Explanation of `copy-translation`

Right now this gulp task automates the creation of these translated `content` and `layout` files by copying them to a new `registrar/` directory within each respective directory.

#### Layout copying ( _straigh-forward_ )
For the `layout` copying, the files are a one-to-one duplicate of the other layouts. Since layout / `html` changes should effect both languages equally, there isn't a need to modify the contents of the files.


#### Content copying ( _less straight-forward_ )
For the `content` copying, the files need to be manipulated in two ways.

1. file names need to be updated to reflect new translated name without accent characters.
1. file content needs to be updated
    1. `title` needs to be translated
    1. `slug` needs to be added to support the new translated name since the `english` version could be easily translated using `{{ replace .Title "." "" | lower | urlize }}` because none of the `english` names have accent characters.

##### How do I update the names in Español?
The updating of the `content` files is done using a `state-names.json` mapping I made manually. Updating the names of states needs to be done in `JSON` at moment, but maybe `toml` or `yaml` would be easier. We can open an issue around it if we need to. The file-mapping translation `JSON` file is within the `config/gulp/` directory because it's only used to generate translated files.